### PR TITLE
feat(tests): added page-object model example. Refactor login test to page-object model. Added a new test channel.spec.ts

### PR DESCRIPTION
**Pull request by nestoragent**
feat(tests): added page-object model example. Refactor login test to page-object model. Added a new test channel.spec.ts

# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: tests/sanity/.eslintrc.js [hunk 1]
```diff
@@ -3,5 +3,8 @@ module.exports = {
   parserOptions: {
     tsconfigRootDir: __dirname,
     project: './tsconfig.json'
+  },
+  rules: {
+    '@typescript-eslint/semi': ['off']
```
Comment (BykhovDenis) [inline tests/sanity/.eslintrc.js]: Why?

**Comment by aplatoff** [comment]
@nestoragent Thank you, Alex. Could you please explain reasons behind switching eslint rule off, and please could you sign your commit so DCO will pass?

**Comment by nestoragent** [comment]
> @nestoragent Thank you, Alex. Could you please explain reasons behind switching eslint rule off, and please could you sign your commit so DCO will pass?

Nothing special behind it.
I'm just very used to them.
But agree with you, better to use the same way for the whole project. 
Will remote the rule and update MR.

**Comment by nestoragent** [comment]
@aplatoff fixed. Thank you

File: tests/sanity/tests/channel.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { generateId, PlatformUser } from './utils'
+import { LoginPage } from './model/login-page'
+import { SelectWorkspacePage } from './model/select-workspace-page'
+import { LeftSideMenuPage } from './model/left-side-menu-page'
+import { ChunterPage } from './model/chunter-page'
+import { ChannelPage } from './model/channel-page'
+
+test.describe('channel tests', () => {
+  test('create new private channel tests', async ({ page }) => {
+    const loginPage = new LoginPage(page)
```
Comment (haiodo) [inline tests/sanity/tests/channel.spec.ts]: Why not open a chunter from beginning why we should have extra step to login and select workspace? I support this is not necessary to perform every time. Please check other tests, some of them open a required page from beginning. Thanks,

**Comment by haiodo** [comment]
In general I like how it goes :-) Nice work.

File: tests/sanity/tests/channel.spec.ts (hunk 2 above)
Comment (nestoragent) [reply tests/sanity/tests/channel.spec.ts]: yep. this isn't necessary.
Will update the test. 
Thank you!

Comment (nestoragent) [reply tests/sanity/tests/channel.spec.ts]: updated.
cc @haiodo

**Review by haiodo — APPROVED** [review]

**Comment by nestoragent** [comment]
Created a new https://github.com/hcengineering/anticrm/pull/3847 with signed commits.
@haiodo review it, please.

**Pull request closed without merging**